sympy/core/mul.py

```diff
@@ -1733,9 +1733,16 @@ def ndiv(a, b):
         # rv will be the default return value
         rv = None
         n, d = fraction(self)
+        n1, d1 = fraction(old)
+        n2, d2 = fraction(new)
         self2 = self
         if d is not S.One:
             self2 = n._subs(old, new)/d._subs(old, new)
+            # Following code handles expressions of the form (d*(a + b)/c).subs((a + b)/c, x) -> d*x
+            # which are often encountered (see issue 17225).
+            if self2.expand() == self.expand() and not d1.is_Number and \
```

Comment:
`expand` can be pretty expensive -- is there a better way to do this?

Reply:
This will work to see if the subs is possible:
```python
>>> (2*(x + y)/z)
(2*x + 2*y)/z
>>> _.extract_multiplicatively((x + y)/z)
2
```

Reply:
Nope expand isn't required , I had some cases in mind where I felt it would be needed but it eventually it worked fine !